File: test_app.py
"""
Fake Python app — used to manually smoke-test DebuggingAI.

Try these in the Extension Development Host after pressing F5:
  1. Open this file
  2. Run "DebuggingAI: Set Breakpoint" on line 22 (process)
  3. Run "DebuggingAI: Set Breakpoint" on line 31 (total) with condition: total > 100
  4. Run "DebuggingAI: List Breakpoints" — check Output Channel
  5. From terminal: python test_app.py
  6. Observe breakpoints hit in VS Code gutter
"""

def add(a, b):
    result = a + b      # bp candidate: inspect a, b, result
    return result


def process(items):
    total = 0
    for item in items:
        total = add(total, item)   # ← set bp here
    return total


def main():
    data = [10, 20, 30, 50]
    total = process(data)          # ← set conditional bp: total > 100
    print(f"Total: {total}")
    assert total == 110, f"expected 110, got {total}"
    print("OK")


if __name__ == "__main__":
    main()

File: test_app.ts
/**
 * Fake TypeScript app — language-agnostic debugger test fixture.
 *
 * Mirrors test_app.py exactly (same structure, same data, same assertion).
 * Use "Debug test_app.ts" launch config to debug this file (via ts-node).
 *
 *   1. Run "DebuggingAI: Set Breakpoint" on line 21 (add call inside process)
 *   2. Run "DebuggingAI: Set Breakpoint" on line 27 (total call) with condition: total > 100
 *   3. Launch "Debug test_app.ts" and observe breakpoints hitting
 */

function add(a: number, b: number): number {
  const result: number = a + b; // bp candidate: inspect a, b, result
  return result;
}

function process(items: number[]): number {
  let total = 0;
  for (const item of items) {
    total = add(total, item); // ← set bp here
  }
  return total;
}

function main(): void {
  const data: number[] = [10, 20, 30, 50];
  const total: number = process(data); // ← set conditional bp: total > 100
  console.log(`Total: ${total}`);
  console.assert(total === 110, `expected 110, got ${total}`);
  console.log('OK');
}

main();
